help/share-and-upload-files.md

```diff
@@ -1,21 +1,103 @@
+the message, if it can. Image thumbnails will be shown directly in the message,
+and you can click on a thumbnail to [view the full image](/help/view-and-browse-images).
 
-* **Drag and drop** files into the compose box.
-* **Copy and paste** files into the compose box.
-* Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
-  the bottom of the compose box to find files on your computer.
+## Uploading files
 
-Zulip will insert a link to the file, in Markdown format:
-`[link text](URL)`. You can modify the `link text` to whatever you want.
+{start_tabs}
 
-Zulip will automatically generate a **thumbnail** of the file when you send
-it, if it can. You can
-[preview the message](/help/preview-your-message-before-sending) before
-sending to see what the thumbnail will look like.
+{tab|via-markdown}
+
+{!start-composing.md!}
+
+1. Drag and drop files, or copy and paste one or more files into the compose
+   box. Zulip will upload the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[Link text](URL)`.
+
+1. _(optional)_ Modify the link text as desired.
+
+!!! tip ""
+
+    You can [preview the message](/help/preview-your-message-before-sending)
+    before sending to see what your uploaded files will look like.
+
+{tab|via-compose-box-buttons}
+
+{!start-composing.md!}
+
+1. Click the **paperclip** (<i class="fa fa-paperclip"></i>) icon at
+   the bottom of the compose box to select one or more files. Zulip will upload
+   the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[Link text](URL)`.
+
+1. _(optional)_ Modify the link text as desired.
+
+!!! tip ""
+
+    You can [preview the message](/help/preview-your-message-before-sending)
+    before sending to see what your uploaded files will look like.
+
+{tab|mobile}
+
+1. Navigate to a stream, topic, or direct message view.
+
+1. Tap the
+   **paperclip** (<img src="/static/images/help/mobile-paperclip-icon.svg" alt="paperclip" class="mobile-icon"/>),
+   **image** (<img src="/static/images/help/mobile-image-icon.svg" alt="image" class="mobile-icon"/>),
+   or **camera** (<img src="/static/images/help/mobile-camera-icon.svg" alt="camera" class="mobile-icon"/>)
+   button at the bottom of the app to select one or more files. Zulip will
+   upload the files, and insert named links using
+   [Markdown formatting](/help/format-your-message-using-markdown#links):
+   `[Link text](URL)`.
+
+1. _(optional)_ Modify the link text as desired.
+
+{end_tabs}
+
+!!! tip ""
+
+    The link text will default to the name of the uploaded file.
+
+## Sharing files
+
+You can share files from other apps on Zulip.
+
+{start_tabs}
+
+{tab|android}
+
+1. Select one or more files and tap the **Zulip**
+   (<img src="/static/images/logo/zulip-icon-circle.svg" alt="logo" class="mobile-icon"/>)
+   logo.
```

Comment:
This is probably clearest if expanded to a couple of steps:

1. In the app that has the item you want to share, find the item and tap **Share** ![image](https://github.com/zulip/zulip/assets/28173/26553858-9ea6-4957-889f-4018c7dbeb2c) to open the sharing menu.

2. In the sharing menu, select **Zulip** <img src="https://github.com/zulip/zulip/assets/28173/2b63c98a-f6e4-4fc1-82b9-a9bed5ea96e7" height=18px />.

That "share" icon is [this one](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:share:FILL@0;wght@400;GRAD@0;opsz@48&icon.query=share&icon.platform=web), the `share` icon in the Material icon set. Fortunately this is one area where Android apps seem to be pretty consistent with each other.

Reply:
I also spent a bit of time looking for anything in Android's own user-facing documentation that we could link to. I didn't; the closest match is this:
https://support.google.com/android/answer/9110661
which is about finding files in the file-browser app, and mentions sharing them from there.